flink-connector-debezium/src/main/java/com/alibaba/ververica/cdc/debezium/DebeziumSourceFunction.java

```diff
@@ -160,6 +161,13 @@
     private static final ConcurrentMap<String, Collection<SchemaRecord>> HISTORY =
             new ConcurrentHashMap<>();
 
+    /**
+     * The schema history will be clean up once {@link DatabaseHistory#stop()}, the checkpoint
+     * should fail when this happens.
+     */
+    private static final ConcurrentMap<String, Boolean> HISTORY_CLEANUP_STATUS =
```

Comment:
Could you move `HISTORY` and `HISTORY_CLEANUP_STATUS` and the accessing methods into another class, and just expose the getter/setters? In this way, we can avoid users/devs to access the underlying  `HISTORY` without lock protect.